### PR TITLE
test: output_spec

### PR DESCRIPTION
**Pull request by justinmk**
Reference: https://github.com/neovim/neovim/issues/3772#issuecomment-162334005

### CODE REVIEW

**Comment by justinmk** [comment]
QB freebsd64, linux64, and yosemite fail correctly:

```
23:15:38,346 INFO  - # Failure message: ./test/functional/ui/screen.lua:289: Row 3 didn't match.
23:15:38,346 INFO  - # Expected:
23:15:38,346 INFO  - # |~ |
23:15:38,346 INFO  - # |~ |
23:15:38,346 INFO  - # |*[No Name] |
23:15:38,346 INFO  - # |:!echo -n foo; sleep 200 |
23:15:38,346 INFO  - # | |
23:15:38,346 INFO  - # |foo |
23:15:38,346 INFO  - # |-- TERMINAL -- |
23:15:38,346 INFO  - # Actual:
23:15:38,346 INFO  - # |~ |
23:15:38,346 INFO  - # |~ |
23:15:38,346 INFO  - # |*~ |
23:15:38,346 INFO  - # |[No Name] |
23:15:38,346 INFO  - # |:!echo -n foo; sleep 200 |
23:15:38,346 INFO  - # | |
23:15:38,346 INFO  - # |-- TERMINAL -- |
23:15:38,346 INFO  - # stack traceback:
23:15:38,346 INFO  - #     ./test/functional/ui/screen.lua:289: in function 'wait'
23:15:38,346 INFO  - #     ./test/functional/ui/screen.lua:221: in function 'expect'
23:15:38,346 INFO  - #     test/functional/ui/output_spec.lua:30: in function <test/functional/ui/output_spec.lua:26>
23:15:38,346 INFO  - #
```

**Comment by justinmk** [comment]
After merging master, QB now succeeds, confirming that #4646 fixes the test.

**Comment by bfredl** [comment]
This hangs for me locally.

**Comment by bfredl** [comment]
It also hangs when testing manually with `nvim -u NORC`, both zsh and bash: `!printf foo; sleep 200`. I need to press `^C` like, four times, before nvim unhangs.

**Comment by justinmk** [comment]
How long did you wait?

File: test/functional/ui/output_spec.lua [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      ~                                                 |
+      ~                                                 |
+      [No Name]                                         |
+                                                        |
+      -- TERMINAL --                                    |
+    ]])
+  end)
+
+  after_each(function()
+    screen:detach()
```
Comment (justinmk) [inline test/functional/ui/output_spec.lua]: @bfredl try removing this line

**Comment by bfredl** [comment]
more than 200 milliseconds. But now I see it is 200 seconds. It sounds a bit excessive, compare the timeout for screen tests that is 250 milliseconds.

**Comment by justinmk** [comment]
The sleep is intentionally excessive. The screen test should timeout even if the sleep were infinite, right? Note the sleep is done in a child nvim inside a `:term` buffer.

I did notice however the screen timeout doesn't seem to work reliably, but that strikes me as a bug in the screen module?

**Comment by justinmk** [comment]
Actually I am pretty sure the detach line is the problem. That would explain why busted reports  a very short time for the actual test.

**Comment by justinmk** [comment]
I think oni link has a PR which would fix that.

**Comment by bfredl** [comment]
I was just thinking that some test target would trigger the same code path as for me and needlessly prolong the build time with 3 minutes, but yeah it is better to fix the actual bug than shortening the time.

**Comment by bfredl** [comment]
> That would explain why busted reports a very short time for the actual test.

How is this a problem? A _successful_ screen test has no lower bound how long time it would take (otherwise adding a lot of screen tests would not scale).

**Comment by justinmk** [comment]
@bfredl I didn't mean to imply that is a problem--only that I was confused about why busted reported ~50ms for the test even though running the test took minutes.

**Comment by justinmk** [comment]
Did removing `screen:detach()` solve the slowness for you?

**Comment by bfredl** [comment]
No. removing `screen:detach()` doesn't change anything. (but lowering the sleep time to 2 seconds works as expected)

**Comment by justinmk** [comment]
well _some_ teardown must be waiting for that `sleep` to finish. I'll try  #4798 later tonight unless you have a chance to try it before then.

**Comment by justinmk** [comment]
@bfredl Just confirmed that https://github.com/neovim/neovim/pull/4798 fixes the "hang". I'll work on getting that merged.

**Pull request merged**